src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -83,185 +103,196 @@ internal struct NameDeclarationInfo
                 CancellationToken cancellationToken, out NameDeclarationInfo result)
             {
-                result = IsLastTokenOfType<ExpressionStatementSyntax>(
+                return IsLastTokenOfType<ExpressionStatementSyntax>(
                     token, semanticModel,
                     e => e.Expression,
                     _ => default(SyntaxTokenList),
                     _ => ImmutableArray.Create(SymbolKind.Local),
-                    cancellationToken);
-                return result.Type != null;
+                    cancellationToken,
+                    out result);
             }
 
             private static bool IsPropertyDeclaration(SyntaxToken token, SemanticModel semanticModel,
                 int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
             {
-                result = IsLastTokenOfType<PropertyDeclarationSyntax>(
+                return IsLastTokenOfType<PropertyDeclarationSyntax>(
                     token,
                     semanticModel,
                     m => m.Type,
                     m => m.Modifiers,
                     GetPossibleDeclarations,
-                    cancellationToken);
-
-                return result.Type != null;
+                    cancellationToken,
+                    out result);
             }
 
             private static bool IsMethodDeclaration(SyntaxToken token, SemanticModel semanticModel,
                 int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
             {
-                result = IsLastTokenOfType<MethodDeclarationSyntax>(
+                return IsLastTokenOfType<MethodDeclarationSyntax>(
                     token,
                     semanticModel,
                     m => m.ReturnType,
                     m => m.Modifiers,
                     GetPossibleDeclarations,
-                    cancellationToken);
-
-                return result.Type != null;
+                    cancellationToken,
+                    out result);
             }
 
-            private static NameDeclarationInfo IsFollowingTypeOrComma<TSyntaxNode>(SyntaxToken token,
+            private static bool IsFollowingTypeOrComma<TSyntaxNode>(SyntaxToken token,
                 SemanticModel semanticModel,
                 Func<TSyntaxNode, SyntaxNode> typeSyntaxGetter,
                 Func<TSyntaxNode, SyntaxTokenList?> modifierGetter,
                 Func<DeclarationModifiers, ImmutableArray<SymbolKind>> possibleDeclarationComputer,
-                CancellationToken cancellationToken) where TSyntaxNode : SyntaxNode
+                CancellationToken cancellationToken,
+                out NameDeclarationInfo result) where TSyntaxNode : SyntaxNode
             {
                 if (!IsPossibleTypeToken(token) && !token.IsKind(SyntaxKind.CommaToken))
                 {
-                    return default;
+                    result = default;
```

Comment:
Might be contentious, but you could just do the "return = default" at the top of the method and not repeat it everywhere.

Reply:
Cyrus ingrained this into me too effectively. Can't do it :(.